### ISSUE TITLE
Category Blog Pagination Incorrect

### ISSUE THREAD

**Issue by Paladin** [joomla/joomla-cms#18541]
### Steps to reproduce the issue
In a website with multiple articles in a given category:

1. Create a new menu item for category blog view of that category.
2. On the Blog Layout tab, adjust the numbers of leading and intro articles so the total is less than the total articles in category (if 5 articles in category, e.g., use 1 leading and 2 intro articles).
3. Set the number of links to any number greater than 1 (try 5 in the above example).
4. View page.
5. click pagination to go to next page.

### Expected result
The articles on the next page begin with the next article after the last linked article.

### Actual result
The articles on the next page begin with the first linked article. IOW, articles linked to on page 1 appear at the top of page 2.

### System information (as much as possible)
Joomla 3.8.1

### Additional comments

It's a User Confusion bug. I ran into it when deciding to override the formatting of the links list (turning it into a grid, if it matters, with a thumbnail of the article image as part of the grid). I had 7 articles, so set the Lead to 1, the Intro to 1, and the Links to 6 to test the new display. And I saw pagination for multiple pages, when every article in the category was already listed on the screen in front of me?!?! I checked and they were repeating articles that had already been presented to the user on a previous page. I don't think that would be the behavior of this layout expected by most users. It certainly wasn't what I expected. When I click "Next page" I expect to see more articles, not a repetition of arcticles I've already decided I didn't want to read (or worse, just finished reading). It appears the number of links that will be shown the user isn't taken into account by the code that calculates number of items per page.

**Comment by brianteeman** [comment]
This annoys me as well but it is the expected behaviour and in tests with my clients they all seem to expect the current behaviour

**Comment by brianteeman** [comment]
I am going to close this for the reasons stated above AND because it has worked this way since the very beginning

**Comment by Paladin** [comment]
TBH, I kind of expected a "Won't fix" response on this, Brian (though I dared hope for something better). Congratulations on your clients; the reaction here when I explained that this was the way it was designed to work was "Well, can't you fix it?" and they weren't especially happy with "Not without patching core" as a response. I would expect that reaction from anyone who hadn't seen/worked with a Joomla site before; Like I said, I envy you not having run into that before.

I definitely agree with continuing to maintain this as default behavior, for B/C reasons, but the way it's hard-coded into ListModel doesn't seem to leave a convenient way of overriding that behavior even with a plugin -- pretty much end up moving a good chunk of the view.html.php code into the template override. I continue to see that as a defect, but I understand closing this.
